src/subcommand/merge_subcommand.cpp

```diff
@@ -54,6 +57,56 @@ void perform_fastforward(repository_wrapper& repo, const git_oid target_oid, int
     target_ref.write_new_ref(target_oid);
 }
 
+void create_merge_commit(repository_wrapper& repo, const index_wrapper& index, std::vector<std::string> m_branches_to_merge,
+    const annotated_commit_list_wrapper& commits_to_merge, size_t num_commits_to_merge)
+{
+    auto head_ref = repo.head();
+    auto merge_ref = repo.find_reference_dwim(m_branches_to_merge.front());
+    auto merge_commit = repo.resolve_local_ref(m_branches_to_merge.front()).value();
+
+    std::vector<commit_wrapper> parents_list;
+    parents_list.reserve(num_commits_to_merge + 1);
+    parents_list.push_back(std::move(head_ref.peel<commit_wrapper>()));
+    for (size_t i=0; i<num_commits_to_merge; ++i)
+    {
+        parents_list.push_back(repo.find_commit(commits_to_merge[i].oid()));
+    }
+    auto parents = commit_list_wrapper(std::move(parents_list));
+
+    auto author_committer_sign = signature_wrapper::get_default_signature_from_env(repo);
+    std::string author_name;
+    author_name = author_committer_sign.first.name();
+    std::string author_email;
+    author_email = author_committer_sign.first.email();
+    auto author_committer_sign_now = signature_wrapper::signature_now(author_name, author_email, author_name, author_email);
+
+    // TODO: add a prompt to edit the merge message
+    std::string msg_target = "";
+    if (merge_ref)
+    {
+        msg_target = merge_ref->short_name();
+    }
+    else
+    {
+        msg_target = git_oid_tostr_s(&(merge_commit.oid()));
+    }
+
+	std::string msg = "Merge ";
+	if (merge_ref)
+	{
+	    msg.append("branch ");
+	}
+	else
+	{
+	    msg.append("commit ");
+	}
```

Comment:
These two "if/else" sections can be merged into a single one. Or you could use the ternary operator to initilaze the message in a single line:

```cpp
std::string msg = merge_ref ? "Merge branch" : "Merge commit'";
```